.github/copilot-instructions.md

```diff
@@ -0,0 +1,254 @@
+# GitHub Copilot Instructions for Application Insights JavaScript SDK
+
+## Project Overview
+This is the **Microsoft Application Insights JavaScript SDK** - a browser-based telemetry library for monitoring web applications. The SDK tracks page views, user interactions, performance metrics, exceptions, and custom events.
+
+## Architecture & Structure
+
+### Monorepo Structure
+- **AISKU/**: Main Application Insights SDK package
+- **AISKULight/**: Lightweight version of the SDK
+- **shared/**: Core shared libraries (AppInsightsCore, AppInsightsCommon, 1ds-core-js)
+- **extensions/**: Plugin-based extensions (analytics, dependencies, React, Angular, etc.)
+- **channels/**: Data transmission channels (online, offline, tee)
+- **tools/**: Build and development tools
+- **examples/**: Sample implementations
+
+### Key Technologies
+- **TypeScript/JavaScript**: Primary languages (ES5 target for browser compatibility)
+- **Rush**: Monorepo management tool
+- **Rollup**: Module bundler
+- **Grunt**: Task runner
+- **Dynamic Proto**: Dynamic prototype pattern for performance
+
+## Code Style & Patterns
+
+### TypeScript/JavaScript Conventions
+- Use **ES5-compatible** syntax for browser support
+- Prefer `function` declarations over arrow functions for better IE compatibility
+- Use `var` instead of `let/const` in some contexts for ES5 compatibility
+- Always use semicolons
+- Use 4-space indentation
+- Maximum line length: 140 characters
+
+### Naming Conventions
+- **Classes**: PascalCase (e.g., `PageViewManager`, `TelemetryContext`)
+- **Interfaces**: PascalCase with `I` prefix (e.g., `ITelemetryItem`, `IPageViewTelemetry`)
+- **Methods/Functions**: camelCase (e.g., `trackPageView`, `sendTelemetry`)
+- **Constants**: UPPER_SNAKE_CASE (e.g., `MAX_DURATION_ALLOWED`)
+- **Private variables**: underscore prefix (e.g., `_logger`, `_hasInitialized`)
```

Comment:
We should add an entry for enums stating that they should be constant and named as PascalCase with a leading `e` and if required to be exported it should use the `createEnumStyle` helper to create the export. And ALL usages within the code base should use the constant enum -- which should also be interers and not strings.